### PR TITLE
Implement a DCAT-AP converter using RDF

### PR DESCRIPTION
**Pull request by jahow**
### Description

This PR introduces a new metadata converter for the DCAT-AP profile. The Metadata Converter app has been modified to allow handling this new format.

Limitations:
* writing to DCAT-AP is very limited for now
* ~reading is limited to certain fields of the model (i.e. no extents, no licence/constraints, etc.)~
* Certain fields of the model are not read yet

Note that the converter can handle either RDF/XML, JSON-LD or Turtle (both for input and output).

**This also fixes a bug where the original XML of a record was lost when it was modified and saved!**

### Architectural changes

Added DcatApConverter class in libs/api/metadata-converter

### Screenshots

image

### Quality Assurance Checklist

- [x] Commit history is devoid of any _merge commits_ and readable to facilitate reviews
- [x] If **new logic** ⚙️ is introduced: unit tests were added
- [ ] If **new user stories** 🤏 are introduced: E2E tests were added
- [ ] If **new UI components** 🕹️ are introduced: corresponding stories in Storybook were created
- [ ] If **breaking changes** 🪚 are introduced: add the `breaking change` label
- [ ] If **bugs** 🐞 are fixed: add the `backport <release branch>` label
- [ ] The [documentation website](docs) 📚 has received the love it deserves

### CODE REVIEW

File: apps/metadata-converter/src/app/app.component.html [hunk 1]
```diff
@@ -11,7 +11,7 @@ <h1 class="text-[50px] font-bold font-title my-6 mx-10">
         ></gn-ui-status>
       </h1>
     </div>
-    <div class="grow p-3">
+    <div class="grow shrink-0 p-3 min-w-[50%]">
       <input
         type="file"
         accept=".xml, application/xml"
```
Comment (tkohr) [inline apps/metadata-converter/src/app/app.component.html:17]: We should probably add `jsonId` and `ttl` here now.

**Review by tkohr — COMMENTED** [review]
Thanks for the big work @jahow ! Code LGTM as far as I can see.

When doing some quick testing with https://github.com/w3c/dxwg/blob/gh-pages/dcat/examples/basic-example.jsonld or ttl in the converter app I get errors when I switch to the ISO formats.

Error for `ttl`:  `Failed: Could not determine role code from role: pointOfContact`
Error for `jsonId`: `Failed: Invalid time value`

**Comment by jahow** [comment]
Thank you for trying things out! I'll look into it

**Review by tkohr — APPROVED** [review]
Thanks, the conversion of the `jsonld` file above now works in the converter. Still getting an error for the same example in `ttl` though: `status.component.ts:90 TypeError: Cannot read properties of undefined (reading 'name') at write-parts.ts:222:39`. Watch out the CI also turned red.

**Pull request merged**